RealmSwift-swift2.2/Tests/RealmTests.swift

```diff
@@ -255,6 +255,29 @@ class RealmTests: TestCase {
         }
         XCTAssertEqual(try! Realm().objects(SwiftStringObject.self).count, 0)
     }
+    
+    func testThrowsWrite() {
+        enum TestError: ErrorType { case Exception }
+        
+        do {
+            try Realm().write {
+                throw TestError.Exception
+            }
+            XCTFail("Expected exception thrown in write block to be rethrown.")
+        }
+        catch TestError.Exception { }
```

Comment:
I realize we have an `assertThrows` function, but it wasn't working for me. Does using `try!` turn a Swift exception into an Objective-C exception? Should I expect this to work on Swift error types, or only `NSError`?